src/test/java/com/iexec/common/utils/TargetedLockRunnerTest.java

```diff
@@ -0,0 +1,89 @@
+package com.iexec.common.utils;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+
+class TargetedLockRunnerTest {
+    /**
+     * Run an action {@code numberOfRunsPerGroup * numberOfActions} times.
+     * <br>
+     * There are {@code numberOfActions} groups of actions, every action in a group having the same input.
+     * Every group is run {@code numberOfRunsPerGroup} times.
+     *
+     * @param <K> Type of the lock key.
+     */
+    private <K> void runWithLock(int numberOfRunsPerGroup,
+                                 int numberOfActions,
+                                 Function<Integer, K> keyProvider) {
+        final TargetedLockRunner<K> locks = new TargetedLockRunner<>();
+
+        final ConcurrentLinkedQueue<Integer> actionsOrder = new ConcurrentLinkedQueue<>();
+        // Run the given action a bunch of times.
+        final Consumer<Integer> runActionAndGetOrder = (i) -> {
+            for (int j = 0; j < numberOfRunsPerGroup; j++) {
+                actionsOrder.add(i);
+            }
+        };
+
+        // Run the action a bunch of times with different inputs and potentially different keys each time.
+        IntStream.range(0, numberOfActions)
+                .parallel()
+                .forEach(i -> locks.runWithLock(keyProvider.apply(i), () -> runActionAndGetOrder.accept(i)));
```

Comment:
Can we check the other methods ? acceptWithLock, getWithLock, applyWithLock, ...

Reply:
We could probably test them, yes. But the test algorithm is already really tricky, I'm not sure I want to add tests for such similar methods. Let me think about this.